src/syntax/ast.ts

```diff
@@ -62,6 +62,18 @@ export interface BeginStatement extends PGNode {
     deferrable?: boolean;
 }
 
+export interface SetTransactionSnapshot extends PGNode {
+    type: 'set transaction snapshot';
+    snapshotId: Name;
+}
+
+export interface SetTransactionMode extends PGNode {
```

Comment:
might make sense to add more granular types? (see suggestion by @oguimbal in https://github.com/oguimbal/pgsql-ast-parser/issues/76)

Reply:
yes, it should: this typing suggests that `set session characteristics as transaction deferrable , isolation level read uncommitted , isolation level read committed ,  read write` is not a valid statement... but it is 🤷‍♂️  try it

 => The fact that my typing suggestion involves an array of characteristics reflects that